### ISSUE TITLE
[fuzzer] One path fails on map_from_entries()

### ISSUE THREAD

**Issue by darrenfu** [facebookincubator/velox#3581]
### How to repro
Run `velox_expression_fuzzer_test` using arguments: `--logtostderr=1 --velox_fuzzer_enable_complex_types --only map_from_entries --duration_sec 60`

### Expected behavior
Both the common and simple paths are behaving the same.

### Actual behavior
The simple path is not returning any exceptions, but the common path returns an error "map key cannot be null" or "map entry cannot be null".

### System information

Velox System Info v0.0.2
Commit: d0ce34ef35a5187c4fc358d8e2993397b82026f4
CMake Version: 3.25.1
System: Darwin-21.6.0
Arch: arm64
C++ Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
C++ Compiler Version: 14.0.0.14000029
C Compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
C Compiler Version: 14.0.0.14000029
CMake Prefix Path: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk/usr;/Users/dofu/homebrew;/usr/local;/usr;/;/Users/dofu/homebrew/Cellar/cmake/3.25.1;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local
  

### Relevant logs

```bash
/Users/dofu/git/velox/cmake-build-debug/velox/expression/tests/velox_expression_fuzzer_test --logtostderr=1 --velox_fuzzer_enable_complex_types --only map_from_entries --duration_sec 60 --gtest_filter=* --gtest_color=no
Testing started at 3:05 PM ...
I20221221 15:05:20.665223 16456133 FuzzerRunner.h:120] Skipping special form: if
I20221221 15:05:20.666692 16456133 FuzzerRunner.h:120] Skipping special form: coalesce
I20221221 15:05:20.666705 16456133 FuzzerRunner.h:120] Skipping special form: switch
I20221221 15:05:20.667228 16456133 ExpressionFuzzer.cpp:351] Total candidate functions: 1 (2 signatures)
I20221221 15:05:20.667243 16456133 ExpressionFuzzer.cpp:355] Functions with at least one supported signature: 1 (100.00%)
I20221221 15:05:20.667255 16456133 ExpressionFuzzer.cpp:359] Functions with no supported signature: 0 (0.00%)
I20221221 15:05:20.667263 16456133 ExpressionFuzzer.cpp:363] Supported function signatures: 1 (50.00%)
I20221221 15:05:20.667269 16456133 ExpressionFuzzer.cpp:367] Unsupported function signatures: 1 (50.00%)
I20221221 15:05:20.667289 16456133 ExpressionFuzzer.cpp:787] ==============================> Started iteration 0 (seed: 1671663920)
I20221221 15:05:20.667438 16456133 ExpressionFuzzer.cpp:608] Couldn't find any function to return 'ARRAY<ROW<"":DATE,"":TINYINT>>'. Returning a constant instead.
I20221221 15:05:20.668223 16456133 ExpressionVerifier.cpp:80] Executing expression: map_from_entries(null)
I20221221 15:05:20.668241 16456133 ExpressionVerifier.cpp:32] 0 vectors as input:
I20221221 15:05:20.669355 16456133 ExpressionVerifier.cpp:69] All results match.
I20221221 15:05:20.669378 16456133 ExpressionFuzzer.cpp:835] ==============================> Done with iteration 0
I20221221 15:05:20.669401 16456133 ExpressionFuzzer.cpp:787] ==============================> Started iteration 1 (seed: 2566370083)
I20221221 15:05:20.671223 16456133 ExpressionVerifier.cpp:80] Executing expression: map_from_entries(10 elements starting at 130 {{[130->47] 72, 10 elements starting at 210 {16700, 12753, null, 6758, null, ...}}, {[131->468] 107, 10 elements starting at 210 {16700, 12753, null, 6758, null, ...}}, {[132->674] 63, 10 elements starting at 210 {16700, 12753, null, 6758, null, ...}}, {[133->280] 117, 10 elements starting at 210 {16700, 12753, null, 6758, null, ...}}, {[134->505] 17, 10 elements starting at 210 {16700, 12753, null, 6758, null, ...}}, ...})
I20221221 15:05:20.671290 16456133 ExpressionVerifier.cpp:32] 0 vectors as input:
E20221221 15:05:20.671443 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:102, Function:operator(), Expression: !isMapKeyNull map key cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
E20221221 15:05:20.671591 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:102, Function:operator(), Expression: !isMapKeyNull map key cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
E20221221 15:05:20.672560 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:102, Function:operator(), Expression: !isMapKeyNull map key cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
I20221221 15:05:20.672623 16456133 ExpressionFuzzer.cpp:835] ==============================> Done with iteration 1
I20221221 15:05:20.672649 16456133 ExpressionFuzzer.cpp:787] ==============================> Started iteration 2 (seed: 570161899)
I20221221 15:05:20.672703 16456133 ExpressionFuzzer.cpp:608] Couldn't find any function to return 'ARRAY<ROW<"":INTEGER,"":REAL>>'. Returning a constant instead.
I20221221 15:05:20.673471 16456133 ExpressionVerifier.cpp:80] Executing expression: map_from_entries(10 elements starting at 300 {[300->165] [165->41] {[41->288] [288->127] null, [41->165] [165->172] [172->119] 0.5614892840385437}, [301->15] [15->398] {[398->79] null, [398->190] [190->78] [78->158] 0.5614892840385437}, [302->127] [127->155] {[155->265] [265->128] 1012033250, [155->31] [31->119] [119->116] 0.5614892840385437}, [303->247] [247->72] {[72->191] [191->103] 188676803, [72->281] [281->48] [48->342] 0.5614892840385437}, [304->329] [329->120] {[120->304] [304->261] 533535820, [120->129] [129->366] [366->391] 0.5614892840385437}, ...})
I20221221 15:05:20.673529 16456133 ExpressionVerifier.cpp:32] 0 vectors as input:
E20221221 15:05:20.673626 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:99, Function:operator(), Expression: !isMapEntryNull map entry cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
E20221221 15:05:20.673736 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:99, Function:operator(), Expression: !isMapEntryNull map entry cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
E20221221 15:05:20.675770 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:102, Function:operator(), Expression: !isMapKeyNull map key cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
W20221221 15:05:20.675843 16456133 FuzzerToolkit.cpp:112] Two different VeloxExceptions were thrown:
	map entry cannot be null
and
	map key cannot be null
I20221221 15:05:20.675855 16456133 ExpressionFuzzer.cpp:835] ==============================> Done with iteration 2
I20221221 15:05:20.675885 16456133 ExpressionFuzzer.cpp:787] ==============================> Started iteration 3 (seed: 1346541760)
I20221221 15:05:20.675954 16456133 ExpressionFuzzer.cpp:608] Couldn't find any function to return 'ARRAY<ROW<"":SMALLINT,"":INTEGER>>'. Returning a constant instead.
I20221221 15:05:20.676761 16456133 ExpressionVerifier.cpp:80] Executing expression: map_from_entries(10 elements starting at 410 {{392, [410->176] [176->180] [180->368] [368->349] 784437378}, {15852, [411->454] [454->543] [543->37] [37->412] 1519253661}, {2452, [412->533] [533->283] [283->58] [58->419] 18346010}, {28568, [413->223] [223->461] [461->83] [83->88] 425490973}, {29529, [414->443] [443->271] [271->164] [164->201] 814321448}, ...})
I20221221 15:05:20.676795 16456133 ExpressionVerifier.cpp:32] 0 vectors as input:
E20221221 15:05:20.676889 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:102, Function:operator(), Expression: !isMapKeyNull map key cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
E20221221 15:05:20.676978 16456133 Exceptions.h:68] Line: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp:102, Function:operator(), Expression: !isMapKeyNull map key cannot be null, Source: USER, ErrorCode: INVALID_ARGUMENT
E20221221 15:05:20.678033 16456133 FuzzerToolkit.cpp:131] Only one path threw exception:
libc++abi: terminating with uncaught exception of type facebook::velox::VeloxUserError: Exception: VeloxUserError
Error Source: USER
Error Code: INVALID_ARGUMENT
Reason: map key cannot be null
Retriable: False
Expression: !isMapKeyNull
Context: map_from_entries(10 elements starting at 410 {{392, [410->176] [176->180] [180->368] [368->349] 784437378}, {15852, [411->454] [454->543] [543->37] [37->412] 1519253661}, {2452, [412->533] [533->283] [283->58] [58->419] 18346010}, {28568, [413->223] [223->461] [461->83] [83->88] 425490973}, {29529, [414->443] [443->271] [271->164] [164->201] 814321448}, ...}:ARRAY<ROW<"":SMALLINT,"":INTEGER>>)
Top-Level Context: Same as context.
Function: operator()
File: /Users/dofu/git/velox/velox/functions/prestosql/MapFromEntries.cpp
Line: 102
Stack trace:
Stack trace has been disabled.Use --velox_exception_user_stacktrace=true to enable it.

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

**Comment by darrenfu** [comment]
I forked a branch to workaround some existing fuzzer bugs to be able to repro the issue.
Please checkout my branch: https://github.com/darrenfu/velox/tree/map_from_entries_w_fuzzer_fix

**Comment by bikramSingh91** [comment]
@darrenfu Found the issue:
Summary, the code that handles constant input in the implementation for map_from_entries had a bug. Consider the case where the input to map_from_entries is a Constant(Array(ROW<>)). It recognizes that the input is a constant and should only run the function on the constant index, it then tries to create a SelectivityVector that should contain only one valid row at the constant index, however when it creates a LocalSelectivityVector for this purpose it assumes that the new selectivityVector will have no valid rows but unfortunately the LocalSelectivityVector has all rows marked as valid by default, so the function ends up running on all rows on the underlying Array(ROW<>) vector and encounters a row that has a null value as a key.

https://github.com/darrenfu/velox/blob/d0ce34ef35a5187c4fc358d8e2993397b82026f4/velox/functions/prestosql/MapFromEntries.cpp#L44-L46

Solution: 
mark all the rows instead of constant index as invalid 
```
      exec::LocalSelectivityVector singleRow(context, flatIndex + 1);
      singleRow->clearAll();
      singleRow->setValid(flatIndex, true);
      singleRow->updateBounds();

```

To give you more context on how i found the bug:

I noticed that the simplified path was running the function for all 100 input rows but the common path had a selectivity vector of size 12. After diving into the implementation for the simplified path, I noticed that one of the paths flattens the input Constant(Array(ROW<>)) into  Array(ROW<>) by copying the value at index 11 into all 100 rows. Therefore the implementation in map_from_entries does not take the specialized path for constantVectors. I then flattened the vector just before it was fed into map_from_entries in the common path and it worked fine. This led me to believe that the bug was in the specialized path in map_from_entries that deals with constant vector. After walking line by line in a debug session I noticed that the code here https://github.com/darrenfu/velox/blob/d0ce34ef35a5187c4fc358d8e2993397b82026f4/velox/functions/prestosql/MapFromEntries.cpp#L94 was being run on all 12 rows which seemed odd. Thats when i realized the bug in https://github.com/darrenfu/velox/blob/d0ce34ef35a5187c4fc358d8e2993397b82026f4/velox/functions/prestosql/MapFromEntries.cpp#L44-L46
